### PR TITLE
Refactored Video-Component

### PR DESCRIPTION
**Pull request by stkeogie**
Refactored Video-Component #400 . Artist Video will now be displayed before movement videos

example:
artist/Q5593

### CODE REVIEW

File: app/src/app/pages/artist/artist.component.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class ArtistComponent implements OnInit, OnDestroy {
       /** Use data service to fetch entity from database */
       this.artist = await this.dataService.findById<Artist>(artistId, EntityType.ARTIST);
       if (this.artist.videos) {
-        this.uniqueEntityVideos[this.artist.id] = this.artist;
+        this.uniqueEntityVideos[0] = this.artist;
```
Comment (EeveesEyes) [inline app/src/app/pages/artist/artist.component.ts:59]: Using '0' as a key on a map does not guarantee order. Furthermore i don't understand what the purpose of `uniqueVideoLinks` is. I think it would be a better idea to remove the dict and to ensure uniqueness on the list using a Set. Obviously this doesn't work because for some reason the video components takes entities as input instead of bare links in string-format. With those you could then use this `Array.from(new Set(linkArray))` to ensure uniqueness. 

tl;dr: We need a refactoring of the video-component. Or a better work-around.

Comment (jschurse) [reply app/src/app/pages/artist/artist.component.ts]: can be tracked here https://github.com/hochschule-darmstadt/openartbrowser/issues/400

**Comment by jschurse** [comment]
@EeveesEyes shall we merge this now, or wait for the refactor #400 ?

**Comment by stkeogie** [comment]
@jschurse  The Pull-Request should be merged after the Refactoring.

File: app/src/app/pages/artist/artist.component.ts [hunk 2]
```diff
@@ -37,27 +37,23 @@ export class ArtistComponent implements OnInit, OnDestroy {
...
-  /* List of unique Videolinks */
-  uniqueVideosLinks: string[];
+  uniqueEntityVideos: any[] = [];
 
   constructor(private dataService: DataService, private route: ActivatedRoute) {}
 
   /** hook that is executed at component initialization */
   ngOnInit() {
     this.route.params.subscribe(() => {
       this.videoExists = false;
```
Comment (EeveesEyes) [inline app/src/app/pages/artist/artist.component.ts:47]: note that `this.videoExists` is already false (line 38)

File: app/src/app/pages/artwork/artwork.component.html [hunk 3]
```diff
@@ -39,10 +39,10 @@
           <app-abstract [entity]="artwork"></app-abstract>
 
           <div class="video">
-            <app-video [entity]="artwork" (videoFound)="videoFound($event)"></app-video>
-            <app-video *ngFor="let artist of artwork.artists.slice(0, 1)" [entity]="artist"
+            <app-video [videoUrl]= "artwork.videos && artwork.videos.length > 0 ? artwork.videos[0] : null" (videoFound)="videoFound($event)"></app-video>
```
Comment (EeveesEyes) [inline app/src/app/pages/artwork/artwork.component.html:42]: use `*ngIf`to hide app-video if of missing artwork video.
(applies to all of the following `artwork.videos && artwork.videos.length > 0 ?`)

Furthermore, why use another method here than in the artist-component. To keep it consistent, we should / could use a uniqueEntityVideos-list here, too. One way to avoid this code duplication would be to move this into a videos-component which manages uniqueEntityVideos-lists and whatnot. @jschurse and @mxxbe what do you think about this?

File: app/src/app/pages/artwork/artwork.component.html [hunk 4]
```diff
@@ -63,7 +63,8 @@
           <app-iconclass i18n="Iconography label@@iconography" [iconclasses]="artwork.iconclasses"
                          label="Iconography"></app-iconclass>
 
-          <app-dimensions *ngIf="artwork" [artwork]="artwork"></app-dimensions>
+          <app-dimensions *ngIf="artwork" [artwork]="
+          artwork"></app-dimensions>
```
Comment (EeveesEyes) [inline app/src/app/pages/artwork/artwork.component.html:67]: new line intended? :)

File: app/src/app/pages/movement/movement.component.html [hunk 5]
```diff
@@ -8,8 +8,8 @@
           <app-abstract [entity]="movement"></app-abstract>
 
           <div class="video">
-            <app-video [entity]="movement" (videoFound)="videoFound($event)"></app-video>
-            <app-video *ngFor="let artwork of sliderItems.slice(0, 10)" [entity]="artwork" (videoFound)="videoFound($event)"></app-video>
+            <app-video [videoUrl]= "movement.videos && movement.videos.length > 0 ? movement.videos[0] : null" (videoFound)="videoFound($event)"></app-video>
```
Comment (EeveesEyes) [inline app/src/app/pages/movement/movement.component.html:11]: same, use `*ngIf` here

File: app/src/app/pages/artist/artist.component.ts (hunk 2 above)
Comment (stkeogie) [reply app/src/app/pages/artist/artist.component.ts:47]: this.videoExists is used to indicate if a video-symbol is shown next to the More-Button. If one of the video-components returns true the Symbol is displayed.

Comment (EeveesEyes) [reply app/src/app/pages/artist/artist.component.ts:47]: sure but it is initialized as false in line 38, so there is no reason to do this again.
